src/parser/parser_builder.cpp

```diff
@@ -143,9 +156,19 @@ ParserBuilder& ParserBuilder::withIncludeFile(bool flag) {
   return *this;
 }
 
-ParserBuilder& ParserBuilder::withForcedLogic(const std::string& logic) {
+ParserBuilder& ParserBuilder::withForcedLogic(const std::string& logic)
+{
   d_logicIsForced = true;
-  d_forcedLogic = logic;
+  if (d_hol)
```

Comment:
`d_hol` should impact this method, which talks about the `force-logic` option, the impact of `d_hol` should be to directly call a method LogicInfo::enableHigherOrder.